src/SFA.DAS.EAS.Employer_Financial.Database/StoredProcedures/GetAllTransactionDetailsForAccountByDate.sql

```diff
@@ -56,6 +56,59 @@ AND DateCreated >= @FromDate
 AND DateCreated < @ToDate
 
 
+
+-- We will use this View to get payments that relate to transaction lines and are funding source 1,2,3
+;WITH AllFunds (DateCreated, AccountId, Ukprn, Uln, PeriodEnd, PaymentId, FundingSource, Amount, PaymentMetaDataId, TransactionTypeDesc,
+	TrainingProvider,
+	Apprentice,
+	ApprenticeTrainingCourse,
+	ApprenticeTrainingCourseLevel)
+AS
+(
+--	SELECT PaymentId, FundingSource, Amount, p.AccountId, p.Ukprn, p.Uln, p.PeriodEnd, p.PaymentMetaDataId
+	SELECT 
+			transLine.DateCreated,
+			p.AccountId, 
+			p.Ukprn, 
+			p.Uln, 
+			p.PeriodEnd, 
+			PaymentId, 
+			FundingSource, 
+			Amount, 
+			p.PaymentMetaDataId, 
+			ptt.[Description] AS TransactionTypeDesc,
+			meta.ProviderName AS TrainingProvider,
+			meta.ApprenticeName AS Apprentice,
+			meta.ApprenticeshipCourseName AS ApprenticeTrainingCourse,
+			meta.ApprenticeshipCourseLevel AS ApprenticeTrainingCourseLevel
+	from [employer_financial].[Payment]   p
+		INNER JOIN (SELECT PeriodEnd,AccountId,ukprn, EmpRef, TransactionDate, DateCreated, LevyDeclared, EnglishFraction 
+					FROM employer_financial.TransactionLine 
+					WHERE DateCreated >= @FromDate AND 
+						DateCreated <= @ToDate) transLine 
+			ON transLine.AccountId = p.AccountId 
+				AND transLine.PeriodEnd = p.PeriodEnd 
+				AND transLine.Ukprn = p.Ukprn
+		INNER JOIN [employer_financial].[PaymentTransactionTypes] ptt 
+			ON ptt.TransactionType =  p.TransactionType
+		INNER JOIN [employer_financial].[PaymentMetaData] meta 
+			ON p.PaymentMetaDataId = meta.Id
+	WHERE  p.AccountId = @AccountId
+		AND DateCreated >= @FromDate 
+		AND DateCreated <= @ToDate
+		AND p.FundingSource in (1,2,3)
+)
+, UniqueApprenticePaymentRecords -- This gets all the unique Account, Provider, Learner, PeriodEnd combinations
+AS
+(
+SELECT AccountId, Ukprn, Uln, PeriodEnd 
+	FROM [employer_financial].[Payment]   p
+		WHERE  p.AccountId = @AccountId
+			AND p.FundingSource in (1,2,3)
+	GROUP BY AccountId, Ukprn, Uln, PeriodEnd
+)
+
+
 INSERT INTO #output
```

Comment:
I can see that the temp table was there before but is there a reason it's needed?